src/components/homepage/calculator-preview.tsx

```diff
@@ -0,0 +1,367 @@
+  const taxBonus = taxBonusAmount > maxTaxBonus ? maxTaxBonus : taxBonusAmount;
+  const netWage = Number(
+    (grossWage - deductions - taxableIncome * incomeTax + taxBonus).toFixed(2)
+  );
+
+  return {
+    netWage: netWage,
+    deductions: deductions,
+    taxableIncome: taxableIncome,
+    nonTaxableAmount: nonTaxableAmount,
+    incomeWithoutDeductions: incomeWithoutDeductions,
+    incomeTaxAmount: incomeTaxAmount,
+    taxBonus: taxBonus,
+  };
+}
+
+//form schema
+const formSchema = z.object({
+  grossWage: z.number().min(minWage, {
+    message: "Gross wage must be at least minimum wage (750€)",
+  }),
+  nonTaxablePart: z.boolean().optional(),
+  childrenUnder18: z
+    .number()
+    .int()
+    .min(0, {
+      message: "Must be at least 0",
+    })
+    .max(10, { message: "Can't be more than 10" }),
+  childrenOver18: z
+    .number()
+    .int()
+    .min(0, {
+      message: "Must be at least 0",
+    })
+    .max(10, { message: "Can't be more than 10" }),
+});
+
+export default function CalculatorPreview() {
+  const [superGrossWage, setSuperGrossWage] = useState(0);
+  const [netWage, setNetWage] = useState(0);
+  const [deductions, setDeductions] = useState(0);
+  const [taxableIncome, setTaxableIncome] = useState(0);
+  const [nonTaxableAmount, setNonTaxableAmount] = useState(0);
+  const [incomeWithoutDeductions, setIncomeWithoutDeductions] = useState(0);
+  const [incomeTaxAmount, setIncomeTaxAmount] = useState(0);
+  const [taxBonus, setTaxBonus] = useState(0);
+
+  const [isVisible, setIsVisible] = useState(false);
+  //form inicialization
+  const form = useForm<z.infer<typeof formSchema>>({
+    mode: "onChange",
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      grossWage: CALCULATOR_CONSTANTS.MIN_WAGE,
+      nonTaxablePart: true,
+      childrenUnder18: 0,
+      childrenOver18: 0,
+    },
+  });
+
+  const handleToggle = () => {
+    setIsVisible(!isVisible);
+  };
+
+  const watchedValues = form.watch();
+
+  // CALCULATE WAGES
+  useEffect(() => {
+    //when wage is lower than min wage dont calculate
+    if (minWage <= watchedValues.grossWage) {
+      const superGrossWage = Number(
+        (watchedValues.grossWage * (1 + companyDeduction)).toFixed(2)
+      );
+      const {
+        netWage,
+        deductions,
+        taxableIncome,
+        nonTaxableAmount,
+        incomeWithoutDeductions,
+        incomeTaxAmount,
+        taxBonus,
+      } = calculateNetWage(
+        watchedValues.grossWage,
+        watchedValues.nonTaxablePart || false,
+        watchedValues.childrenUnder18,
+        watchedValues.childrenOver18
+      );
+      setSuperGrossWage(superGrossWage);
+      setNetWage(netWage);
+      setDeductions(deductions);
+      setTaxableIncome(taxableIncome);
+      setNonTaxableAmount(nonTaxableAmount);
+      setIncomeWithoutDeductions(incomeWithoutDeductions);
+      setIncomeTaxAmount(incomeTaxAmount);
+      setTaxBonus(taxBonus);
+    }
+  }, [watchedValues]);
```

Comment:
A na toto som presne čakal kedy príde tento moment 😄 celý tento súbor obsahuje uplne že milion takých začiatočníckych reactovských chýb, ktoré spravil uplne každý. K tomuto celému sa musime bud stretnut alebo si zavolat lebo toto by som tu napísal román aby som ti to vysvetlil prečo je to zle 😄